kie-server-parent/kie-server-services/kie-server-services-jbpm/src/test/java/org/kie/server/services/jbpm/RuntimeDataServiceBaseTest.java

```diff
@@ -0,0 +1,93 @@
+import org.jbpm.services.api.model.ProcessDefinition;
+import org.junit.Test;
+import org.kie.server.api.model.KieServerConfig;
+import org.kie.server.api.model.definition.ProcessDefinitionList;
+import org.kie.server.services.api.ContainerLocator;
+import org.kie.server.services.api.KieServerRegistry;
+
+public class RuntimeDataServiceBaseTest {
+
+    private final AdvanceRuntimeDataService advanceRuntimeDataService = mock(AdvanceRuntimeDataService.class);
+    private final RuntimeDataService runtimeDataService = mock(RuntimeDataService.class);
+    private final KieServerRegistry context = mock(KieServerRegistry.class);
+    private final KieServerConfig config = mock(KieServerConfig.class);
+
+    public RuntimeDataServiceBaseTest() {
+        when(context.getConfig()).thenReturn(config);
+    }
+
+    @Test
+    public void testGetProcessesByDeploymentId() {
+        RuntimeDataServiceBase svc = new RuntimeDataServiceBase(runtimeDataService, advanceRuntimeDataService, context);
+
+        String containerId = "container_id";
+        when(context.getContainerId(anyString(), any(ContainerLocator.class))).thenReturn(containerId);
+        List<ProcessDefinition> definitions = new ArrayList<>();
+        definitions.add( mock(ProcessDefinition.class));
+        when(runtimeDataService.getProcessesByDeploymentId(anyString(), any())).thenReturn(definitions);
+
+        ProcessDefinitionList result = svc.getProcessesByDeploymentId(containerId, 0, 100, null, true);
+
+        assertEquals(1, result.getItems().size());
+    }
+
+    @Test
+    public void testGetProcessesByDeploymentId_DeploymentNotFound() {
+        RuntimeDataServiceBase svc = new RuntimeDataServiceBase(runtimeDataService, advanceRuntimeDataService, context);
+
+        String containerId = "container_id";
+        when(context.getContainerId(anyString(), any(ContainerLocator.class))).thenReturn(containerId);
+        List<ProcessDefinition> definitions = new ArrayList<>();
+        definitions.add( mock(ProcessDefinition.class));
+        when(runtimeDataService.getProcessesByDeploymentId(anyString(), any())).thenThrow(new DeploymentNotFoundException("not found"));
+
+        ProcessDefinitionList result = svc.getProcessesByDeploymentId(containerId, 0, 100, null, true);
+
+        assertTrue(result.getItems().isEmpty());
+    }
+
+    @Test
+    public void testGetProcessesByDeploymentId_IllegalArgumentException() {
+        RuntimeDataServiceBase svc = new RuntimeDataServiceBase(runtimeDataService, advanceRuntimeDataService, context);
+
+        String containerId = "container_id";
+        when(context.getContainerId(anyString(), any(ContainerLocator.class))).thenReturn(containerId);
+        List<ProcessDefinition> definitions = new ArrayList<>();
+        definitions.add( mock(ProcessDefinition.class));
+        when(runtimeDataService.getProcessesByDeploymentId(anyString(), any())).thenThrow(new IllegalArgumentException("not found"));
```

Comment:
maybe something different:
```suggestion
        when(runtimeDataService.getProcessesByDeploymentId(anyString(), any())).thenThrow(new IllegalArgumentException("illegal argument"));
```

Reply:
when the resource is not found the RuntimeDataService returns an IllegalArgumentException, I consider this error message more appropriate to the test case